keps/sig-storage/1979-object-storage-support/README.md

```diff
@@ -3,703 +3,832 @@
+More information about Bucket is [here](#bucket)
 
-```yaml
-apiVersion: v1
-kind: Pod
-metadata:
-  name: my-app-pod
-  namespace: dev-ns
-spec:
-  serviceAccountName: [1]
-  containers:
-    - name: my-app
-      image: ...
-      volumeMounts:
-        - mountPath: /cosi [2]
-          name: cosi-vol
-  volumes:
-    - name: cosi-vol
-      csi: [3]
-        driver: cosi.sigs.k8s.io [4]
-        volumeAttributes: [5]
-          bucketAccessRequestName: <my-bar-name>
 ```
-1. the service account may be needed depending on cloud IAM integration with kubernetes.
-1. the mount path is the directory where the app will read the credentials and endpoint.
-1. this is an inline CSI volume.
-1. the name of the cosi-node-adapter.
-1. information needed by the cosi-node-adapter to verify that the bucket has been provisioned.
+    Bucket - bcl-$uuid
+    |--------------------------------------|
+    | name: bcl-$uuid                      |
+    | spec:                                |
+    |   bucketClassName: bc1               |
+    |   protocol: s3                       |
+    |   parameters:                        |
+    |     key: value                       |
+    |   provisioner: s3.amazonaws.com      |
+    | status:                              |
+    |   conditions:                        |
+    |   - type: BucketReady                |
+    |     status: False                    |
+    |--------------------------------------|
+```
 
-The contents of the secret generated by the provisioner is expected to have the following fields in the provider specific format:
+###### 3. COSI calls appropriate driver to create Bucket
 
-- Endpoint for the bucket. The Endpoint should encode the URL for the provider
-- Access Key: Only set when serviceAccount is empty 
-- Secret Key: Only set when serviceAccount is empty
+The COSI sidecar, which runs alongside each of the drivers, listens for Bucket events. All but the sidecar with the specified provisioner will ignore the Bucket object. Only the appropriate sidecar will make a gRPC request to create a Bucket in the OSP.
 
-### Topology
+More information about COSI gRPC API is [here](#cosi-grpc-api)
 
-![Architecture Diagram](images/arch.png)
+```
+    COSI Driver (s3.amazonaws.com)
+    |------------------------------------------|
+    | grpc ProvisionerCreateBucket({           |
+    |      "name": "bcl-$uuid",                |
+    |      "protocol": "s3",                   |
+    |      "parameters": {                     |
+    |         "key": "value"                   |
+    |      }                                   |
+    | })                                       |
+    |------------------------------------------|
+```
+Once the Bucket is successfully created in the OSP, and a successful status is retreieved from the gRPC request, sidecar sets `BucketReady`condition to `True`. At this point, the Bucket is ready to be utilized by workloads.
+
+## Generating Access Credentials for Buckets
```

Comment:
Can you give a couple more examples to outline why we want BucketAccess to be decoupled from BucketClaim? For example;
-  a BucketAccessClass that could represent a "read-only" policy and another to represent a "write" policy
- Give one workload read-only access, and another workload write access.

Reply:
Exactly those two actually. I've updated the kep to contain these examples.